api/proto/config/devmodel.proto

```diff
@@ -117,7 +117,14 @@ message PhysicalIO {
       // Example Eth0, Eth1, Wifi0, ComA, ComB
       string phylabel           = 2;
 
-      // the hardware bus address
+      // The hardware bus address. The key to this map can be of the following
+      // (case-insensitive) values:
+      // "pcishort": the address is a PCI id of the form 02:00.0
+      // "pcilong": the address is a PCI id of the form 0000:02:00.0
+      // "ifname": the addresss is a string for a network interface like "eth1"
+      // "serial": the address is a Linux serial port alias such as "/dev/ttyS2"
+      // "irq": the address is a number such as "5"
+      // "ioports": the address is a string such as "2f8-2ff"
       map <string, string> phyaddrs = 3;
```

Comment:
It seems that splitting irq/ioport breaks our model of having this as a series of aliases to a given hardware object. I'd like to understand how this will be handled before we merge this.

Reply:
@roman Are you saying that since irq is itself not an alias for COMx, we need to have a model where <irq, ioports> is an alias for COMx?

Reply:
yup. that's exactly right @nordmark

Reply:
I realize from the discussion this morning that phyaddrs is a mixture of lookup keys and results of the lookups. For instance, it wouldn't make sense to look up using <irq, ioports>, and a future lookup on <vendor, product> might return multiple matches, hence might need some additional phyaddrs to get to a single answer.

I wish this was cleaner, but the reality we need to handle is a bit of a mess IMHO.
@rvs any ideas for how to improve it?